app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedActivityTest.kt

```diff
@@ -0,0 +1,380 @@
+      scrollToPosition(position = 0)
+      verifyStoryNameOnStorySummaryListItemAtPosition(
+        itemPosition = 0,
+        stringToMatch = "First Story"
+      )
+      scrollToPosition(position = 1)
+      verifyStoryNameOnStorySummaryListItemAtPosition(
+        itemPosition = 1,
+        stringToMatch = "Other Interesting Story"
+      )
+      scrollToPosition(position = 2)
+      verifyStoryNameOnStorySummaryListItemAtPosition(
+        itemPosition = 2,
+        stringToMatch = "Matthew Goes to the Bakery"
+      )
+      scrollToPosition(position = 3)
+      verifyStoryNameOnStorySummaryListItemAtPosition(
+        itemPosition = 3,
+        stringToMatch = "Ratios: Part 1"
+      )
+      scrollToPosition(position = 4)
+      verifyStoryNameOnStorySummaryListItemAtPosition(
+        itemPosition = 4,
+        stringToMatch = "Ratios: Part 2"
+      )
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedActivity_chapterListIsVisible() {
+    launch<MarkChaptersCompletedActivity>(
+      createMarkChaptersCompletedActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 0)
+      scrollToPosition(position = 1)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 1)
+      scrollToPosition(position = 2)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 2)
+      scrollToPosition(position = 3)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 3)
+      scrollToPosition(position = 4)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 4)
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedActivity_configChange_chapterListIsVisible() {
+    launch<MarkChaptersCompletedActivity>(
+      createMarkChaptersCompletedActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 0)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 0)
+      scrollToPosition(position = 1)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 1)
+      scrollToPosition(position = 2)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 2)
+      scrollToPosition(position = 3)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 3)
+      scrollToPosition(position = 4)
+      verifyItemDisplayedOnStorySummaryListItem(itemPosition = 4)
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedActivity_selectAll_isChecked() {
+    launch<MarkChaptersCompletedActivity>(
+      createMarkChaptersCompletedActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.mark_chapters_completed_all_check_box))
+        .perform(click()).check(matches(isChecked()))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedActivity_selectAll_unselect_isNotChecked() {
```

Comment:
add 1 more test `testMarkChaptersCompletedActivity_selectAll_unselect_configChange_isNotChecked`

Reply:
This test is now redundant. I have removed it. Will add new tests with their configChange variants.